intrinsic_and_extrinsic_optimise_real_data/src/camera_calibration_real_data.cpp

```diff
@@ -0,0 +1,422 @@
+	double matrix_D_[4][Number_of_internal_corners_y * Number_of_internal_corners_x];
+
+	count = 0;
+	for (int y = 0; y != Number_of_internal_corners_y; y++)
+	{
+		for (int x = 0; x != Number_of_internal_corners_x; x++)
+		{
+			matrix_D_[0][count] = x * Block_dimension;
+			matrix_D_[1][count] = y * Block_dimension;
+			matrix_D_[2][count] = 0 * Block_dimension;
+			matrix_D_[3][count] = 1;
+
+			count++;
+		}	
+	}
+
+	// The multidimensional arrays are converted into OpenCV matrices so that the multiplication can be done.
+	cv::Mat matrix_A[total_chessboards_detected];
+	cv::Mat matrix_B___ = cv::Mat(3, 3, CV_64FC1, matrix_B__);
+	std::vector<cv::Mat> matrix_C_;
+	cv::Mat matrix_D__ = cv::Mat(4, Number_of_internal_corners_y * Number_of_internal_corners_x, CV_64FC1, matrix_D_);
+	
+
+	for (int z = 0; z != total_chessboards_detected; z++)
+	{
+		matrix_C_.push_back(cv::Mat(3, 4, CV_64FC1, &matrix_C[z][0][0]));
+		//std::cout << "matrix_C_: \r\n" << matrix_C_[z];
+	}
+
+	std::cout << "\r\n";
+
+
+	for (int z = 0; z != total_chessboards_detected; z++)
+	{
+		matrix_A[z] = matrix_B___ * matrix_C_[z] * matrix_D__;
+		// std::cout << "matrix_A: \r\n" << matrix_A[z];
+		// std::cout << "\r\n";
+	}
+
+	// Display the real and calculated matrix A here. 
+	double true_matrix_A[total_chessboards_detected][2][Number_of_internal_corners_y * Number_of_internal_corners_x];
+
+	for (int z = 0; z != total_chessboards_detected; z++)
+	{
+		for (int i = 0; i != Number_of_internal_corners_y * Number_of_internal_corners_x; i++)
+		{
+			true_matrix_A[z][0][i] = Chessboard_corners[z][i].x;
+			true_matrix_A[z][1][i] = Chessboard_corners[z][i].y;
+		}
+	}
+
+	double calculated_matrix_A[total_chessboards_detected][3][Number_of_internal_corners_y * Number_of_internal_corners_x];
+
+	for (int z = 0; z != total_chessboards_detected; z++)
+	{
+		for (int y = 0; y != 3; y++)
+		{
+			for (int x = 0; x != Number_of_internal_corners_y * Number_of_internal_corners_x; x++)
+			{
+				calculated_matrix_A[z][y][x] = matrix_A[z].at<double>(y,x);
+			}	
+		}
+	}
+
+
+	// Recreate the images but using the calculated pixel coordinates.
+	std::vector<std::vector<cv::Point2f>> Processed_Chessboard_corners;
+
+	for (int z = 0; z != total_chessboards_detected; z++)
+	{
+		std::vector<cv::Point2f> temp;
+		for (int x = 0; x != Number_of_internal_corners_y * Number_of_internal_corners_x; x++)
+		{
+			double U = calculated_matrix_A[z][0][x] / calculated_matrix_A[z][2][x]; // (s * u) / s
+			double V = calculated_matrix_A[z][1][x] / calculated_matrix_A[z][2][x]; // (s * v) / s
+			temp.push_back(cv::Point2f(U, V));
+		}	
+		Processed_Chessboard_corners.push_back(temp);
+	}
+
+	for (int z = 0; z != total_chessboards_detected; z++)
+	{
+		cv::namedWindow("Optimised Pixel Points");
+		cv::drawChessboardCorners(images[z], patternsize, Processed_Chessboard_corners[z], true);
+		cv::imshow("Optimised Pixel Points", images[z]);
+		cv::waitKey(0); 
```

Comment:
Ideally the app would be return immediately, without waiting on any user input.  Thus, I'd suggest outputting these images to disk instead of calling imshow.